clar2wasm/tests/wasm-generation/contracts.rs

```diff
@@ -569,18 +569,26 @@ proptest! {
             .unwrap(),
         )
         .into();
+        let expected_res_ty = ('a'..).zip(tys.iter()).fold('{'.to_string(), |mut acc, (c, ty)| {
+            write!(acc, "{c}: {}, ", type_string(ty)).unwrap();
+            acc
+        }) + "}";
```

Comment:
nit: isn't this the same as `expected_res.type_string()`?

Reply:
e.g: for a `none` with type `(optional <anything>)`, if we call `type_string` on the value we would have `(optional int)`. We have to use the correct precise type here.

Reply:
> None => "int".to_owned(), // We need to default to something here

Ok I wasn't expecting that, but it makes sense I guess